.eslintrc.js

```diff
@@ -0,0 +1,188 @@
+module.exports = {
+  settings: {
+    react: {
+      version: "detect",
+    },
+    "import/parsers": {
+      "@typescript-eslint/parser": [".ts", ".tsx"],
+    },
+    "import/resolver": {
+      typescript: {
+        alwaysTryTypes: true, // always try to resolve types under `<roo/>@types` directory even it doesn't contain any source code, like `@types/unist`
+      },
+    },
+  },
+  env: {
+    browser: true,
+    es6: true,
+  },
+  extends: [
+    "plugin:@typescript-eslint/recommended",
+    "plugin:@typescript-eslint/recommended-requiring-type-checking",
+    "plugin:jsx-a11y/recommended",
+    "plugin:react/recommended",
+    "plugin:import/errors",
+    "plugin:import/warnings",
+    "plugin:import/typescript",
+    "plugin:relay/recommended",
```

Comment:
Also noticed that there's no eslint-config-prettier plugin setup here, so that might be why so many rules have been impacted below.

Reply:
Most of those I didn’t add manually, but are generated from the tslint to Eslint migration tool. 

But, yeah, still wip

Reply:
We can / should do this by hand, which will be better long-term. Having every rule listed out is a maintenence burden while also being kinda confusing for other devs in the future ("why did x set this in this way?")

Reply:
I agree that having too many rules would be confusing.

This PR isn't final, it's just a good starting point. I used the tool because it's recommended and a good kicking off point to get the same coverage as we had. Optimizations like adding the prettier plugin (which is installed, but I just forgot to add) will likely reduce the amount of rules here. 

Again though, this is just a WIP PR and in no way meant to represent the final thing. Just wanted to get something up.